### PR TITLE
Added periodic synchronization for blacklists

### PR DESCRIPTION
**Pull request by Jakob-em**
See #8 
- Added implementation of new blacklist loader
- Added database for blacklisted hosts
- Added tests
- Updated Readme

### CODE REVIEW

File: server/main.go [hunk 1]
```diff
@@ -1,28 +1,42 @@
...
+	log "github.com/sirupsen/logrus"
 	"net/http"
 	"strings"
+	"time"
+	"unshort.link/blacklist"
 )
 
-//go:generate curl -o blacklist.txt https://hosts.ubuntu101.co.za/domains.list
 //go:generate go get -u github.com/programmfabrik/esc
-//go:generate esc -private -local-prefix-cwd -pkg=main -o=static.go static/ blacklist.txt
```
Comment (simonfrey) [inline server/main.go]: I assume you did not clean build? As if you remove this complete line the `go build` does not work anymore, as the static files are not build into static.go anymore. 
This line should be keept but, blacklist.txt removed::
`//go:generate esc -private -local-prefix-cwd -pkg=main -o=static.go static/` than the generate and build works as expected

File: server/README.md [hunk 2]
```diff
@@ -26,4 +25,6 @@ be up and running
 
 - `--url`: Set the url of the server you are running on (this is only required for the frontend) (Default: `http://localhost:8080`)
 - `--port`: Port to start the server on (Default: `8080`)
-- `--local`: Use the assets (frontend & blacklist) directly from the filesystem instead of the internal binary storage. This helps during the development of the frontend as you do not have to do `go generate ./...` after every change. This should not be used in production. (Default: `false`)
\ No newline at end of file
+- `--local`: Use the assets (frontend & blacklist) directly from the filesystem instead of the internal binary storage. This helps during the development of the frontend as you do not have to do `go generate ./...` after every change. This should not be used in production. (Default: `false`)
+- `--blacklist-sources`:  Comma separated list of blacklist urls to periodically sync (Default https://hosts.ubuntu101.co.za/domains.list)
+- `--sync`: Blacklist synchronization interval (Default: one hour)
```
Comment (simonfrey) [inline server/README.md]: Please add info about the interval time. What does e.g. `--sync 60` mean? 60 milliseconds/seconds/minutes/hours ?

File: server/README.md [hunk 3]
```diff
@@ -26,4 +25,6 @@ be up and running
 
 - `--url`: Set the url of the server you are running on (this is only required for the frontend) (Default: `http://localhost:8080`)
 - `--port`: Port to start the server on (Default: `8080`)
-- `--local`: Use the assets (frontend & blacklist) directly from the filesystem instead of the internal binary storage. This helps during the development of the frontend as you do not have to do `go generate ./...` after every change. This should not be used in production. (Default: `false`)
\ No newline at end of file
+- `--local`: Use the assets (frontend & blacklist) directly from the filesystem instead of the internal binary storage. This helps during the development of the frontend as you do not have to do `go generate ./...` after every change. This should not be used in production. (Default: `false`)
+- `--blacklist-sources`:  Comma separated list of blacklist urls to periodically sync (Default https://hosts.ubuntu101.co.za/domains.list)
```
Comment (simonfrey) [inline server/README.md]: We should document/define a supported format here? That only files that are newline separated domains are supported.

File: server/README.md (hunk 2 above)
Comment (simonfrey) [reply server/README.md]: Ah found the format in the command help. `XhXmXs` please add it here in the documenttion as well

File: server/main.go (hunk 1 above)
Comment (simonfrey) [reply server/main.go]: For reproducing that on your machine: 

1) Remove `static.go` 
2) Run `go generate ./...`
3) Run `go buid`

Comment (Jakob-em) [reply server/main.go]: Good point I will change it

File: server/README.md (hunk 3 above)
Comment (Jakob-em) [reply server/README.md]: Okay i will adjust the documentation

**Pull request merged**